hack/verify-generators.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT="$(dirname "${BASH_SOURCE}")/.."
+DIFFROOT="${SCRIPT_ROOT}"
+
+cleanup() {
+  git checkout "${DIFFROOT}"
```

Comment:
This will overwrite a dirty working copy for a user who runs this, no?

Should we register this trap after the git porcelain check below?

Reply:
Again this is copied from the adjacent script which does a similar thing, but yeah this is a good catch so I fixed it in both scripts in 8e19d6e66c78ebb573423c39763389d2f5366275